### PR TITLE
Fix Issue 22436 - noway_assert  'expTyp == cseLclVarTyp' in optcse.cpp

### PR DESCRIPTION
**Pull request by briansull**
We now allow TYP_BYREF to be compatible with TYP_I_IMPL when performing CSE substitutions.
Re-enable the failing BestFitMapping interop tests

### CODE REVIEW

**Comment by briansull** [comment]
@dotnet-bot test Windows_NT x64 Checked Innerloop jitstress2 Build and Test

**Comment by briansull** [comment]
@dotnet-bot   test Windows_NT x64 Checked jitstress2
@dotnet-bot   test Windows_NT x64 Checked jitstress2_tiered

File: src/jit/optcse.cpp [hunk 1]
```diff
@@ -1971,6 +1971,26 @@ class CSE_Heuristic
         return result;
     }
 
+    // IsCompatibleType() takes two var_types and returns true if they
+    // are compatible types for CSE substitution
+    //
+    bool IsCompatibleType(var_types cseLclVarTyp, var_types expTyp)
+    {
+        // Excat type match is the expected case
```
Comment (AaronRobinsonMSFT) [inline src/jit/optcse.cpp]: ~~Excat~~ -> Exact

**Comment by briansull** [comment]
@AndyAyersMS  Agreed

**Comment by briansull** [comment]
@dotnet-bot test Windows_NT x64 Checked jitstress2
@dotnet-bot test Windows_NT x64 Checked jitstress2_tiered

**Pull request merged**